### ISSUE TITLE
Undefined variable $newBlock

### ISSUE THREAD

**Issue by nickdunn** [spicywebau/craft-neo#500]
#### Description
Upgraded to latest (2.11.10) and I'm clicking around to make sure site is stable. I went to a random entry with a Neo field, saved it and it was fine, then re-saved again and now I consistently get the error:

```
PHP Warning – yii\base\ErrorException
Undefined variable $newBlock

yii\base\ErrorException: Undefined variable $newBlock in /var/www/craft/vendor/spicyweb/craft-neo/src/services/Fields.php:386
Stack trace:
#0 /var/www/craft/vendor/craftcms/cms/src/web/ErrorHandler.php(75): yii\base\ErrorHandler->handleError(2, 'Undefined varia...', '/var/www/craft/...', 386)
#1 /var/www/craft/vendor/spicyweb/craft-neo/src/services/Fields.php(386): craft\web\ErrorHandler->handleError(2, 'Undefined varia...', '/var/www/craft/...', 386)
#2 /var/www/craft/vendor/spicyweb/craft-neo/src/Field.php(696): benf\neo\services\Fields->duplicateBlocks(Object(benf\neo\Field), Object(craft\elements\Entry), Object(craft\elements\Entry), true)
#3 /var/www/craft/vendor/craftcms/cms/src/base/Element.php(4007): benf\neo\Field->afterElementPropagate(Object(craft\elements\Entry), false)
#4 /var/www/craft/vendor/craftcms/cms/src/elements/Entry.php(1670): craft\base\Element->afterPropagate(false)
#5 /var/www/craft/vendor/craftcms/cms/src/services/Elements.php(1317): craft\elements\Entry->afterPropagate(false)
#6 /var/www/craft/vendor/craftcms/cms/src/services/Elements.php(887): craft\services\Elements->duplicateElement(Object(craft\elements\Entry), Array)
#7 /var/www/craft/vendor/craftcms/cms/src/services/Drafts.php(342): craft\services\Elements->updateCanonicalElement(Object(craft\elements\Entry), Array)
#8 /var/www/craft/vendor/craftcms/cms/src/controllers/EntryRevisionsController.php(479): craft\services\Drafts->applyDraft(Object(craft\elements\Entry))
#9 [internal function]: craft\controllers\EntryRevisionsController->actionPublishDraft()
#10 /var/www/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#11 /var/www/craft/vendor/yiisoft/yii2/base/Controller.php(181): yii\base\InlineAction->runWithParams(Array)
#12 /var/www/craft/vendor/craftcms/cms/src/web/Controller.php(190): yii\base\Controller->runAction('publish-draft', Array)
#13 /var/www/craft/vendor/yiisoft/yii2/base/Module.php(534): craft\web\Controller->runAction('publish-draft', Array)
#14 /var/www/craft/vendor/craftcms/cms/src/web/Application.php(276): yii\base\Module->runAction('entry-revisions...', Array)
#15 /var/www/craft/vendor/craftcms/cms/src/web/Application.php(585): craft\web\Application->runAction('entry-revisions...', Array)
#16 /var/www/craft/vendor/craftcms/cms/src/web/Application.php(255): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#17 /var/www/craft/vendor/yiisoft/yii2/base/Application.php(392): craft\web\Application->handleRequest(Object(craft\web\Request))
#18 /var/www/craft/web/index.php(22): yii\base\Application->run()
#19 {main}
```

#### Other information
- Neo version: 2.11.10
- Craft version: 3.7.12

**Comment by nickdunn** [comment]
After upgrading to 2.11.10 from 2.11.4 I noticed that `neoblockstructures` went from 2,542 to 2,150 rows, presumably a result of the cleanup migration.

Around the same time, I have also recently changed my Craft maxRevisions from 25 to 10 locally and run the `prune` CLI task to clean up.

**Comment by ttempleton** [comment]
Thanks for reporting this — it should now be fixed in Neo 2.11.11.

**Comment by nickdunn** [comment]
Cheers. Was this a result of reducing the number of revisions?

**Comment by ttempleton** [comment]
It was unrelated to revisions, it was a bug introduced in Neo 2.11.10. There was a case, when Neo needs to duplicate blocks for a field/entry, where we were trying to access a duplicate block's properties when the duplicate block variable might not have been initialised.

**Comment by nickdunn** [comment]
Thanks! I'm glad someone understands these complexities :-)

**Comment by nickdunn** [comment]
Updated to 2.11.11 and receiving similar:

```
Error: Attempt to assign property "level" on null in /var/www/craft/vendor/spicyweb/craft-neo/src/services/Fields.php:392
Stack trace:
#0 /var/www/craft/vendor/spicyweb/craft-neo/src/Field.php(696): benf\neo\services\Fields->duplicateBlocks(Object(benf\neo\Field), Object(craft\elements\Entry), Object(craft\elements\Entry), true)
#1 /var/www/craft/vendor/craftcms/cms/src/base/Element.php(4007): benf\neo\Field->afterElementPropagate(Object(craft\elements\Entry), false)
#2 /var/www/craft/vendor/craftcms/cms/src/elements/Entry.php(1670): craft\base\Element->afterPropagate(false)
#3 /var/www/craft/vendor/craftcms/cms/src/services/Elements.php(1317): craft\elements\Entry->afterPropagate(false)
#4 /var/www/craft/vendor/craftcms/cms/src/services/Elements.php(887): craft\services\Elements->duplicateElement(Object(craft\elements\Entry), Array)
#5 /var/www/craft/vendor/craftcms/cms/src/services/Drafts.php(342): craft\services\Elements->updateCanonicalElement(Object(craft\elements\Entry), Array)
#6 /var/www/craft/vendor/craftcms/cms/src/controllers/EntryRevisionsController.php(479): craft\services\Drafts->applyDraft(Object(craft\elements\Entry))
#7 [internal function]: craft\controllers\EntryRevisionsController->actionPublishDraft()
#8 /var/www/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#9 /var/www/craft/vendor/yiisoft/yii2/base/Controller.php(181): yii\base\InlineAction->runWithParams(Array)
#10 /var/www/craft/vendor/craftcms/cms/src/web/Controller.php(190): yii\base\Controller->runAction('publish-draft', Array)
#11 /var/www/craft/vendor/yiisoft/yii2/base/Module.php(534): craft\web\Controller->runAction('publish-draft', Array)
#12 /var/www/craft/vendor/craftcms/cms/src/web/Application.php(276): yii\base\Module->runAction('entry-revisions...', Array)
#13 /var/www/craft/vendor/craftcms/cms/src/web/Application.php(585): craft\web\Application->runAction('entry-revisions...', Array)
#14 /var/www/craft/vendor/craftcms/cms/src/web/Application.php(255): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#15 /var/www/craft/vendor/yiisoft/yii2/base/Application.php(392): craft\web\Application->handleRequest(Object(craft\web\Request))
#16 /var/www/craft/web/index.php(22): yii\base\Application->run()
#17 {main}
```

**Comment by ttempleton** [comment]
Yep, that was happening for the same reason. Just released Neo 2.11.12 which should hopefully completely fix it.
